Allow switching between oss and standard packages

Crazybus opened the pull request:
Before only going from the standard package to oss was supported. Now
this works properly in both directions.

Fixes: #520

fedelemantuano commented:
Hi @Crazybus,

if you switch from `oss` to `xpack` or vice versa, you will have two [two yum repo](https://github.com/elastic/ansible-elasticsearch/blob/c2e51f8ecfdcb99c8e2b0b177b372ac7fcd432ee/tasks/elasticsearch-RedHat.yml#L14) in system.
To have a clean config it's better remove the old one.
I know that change only the name, maybe you could use a unique name for repos oss and xpack.

fedelemantuano commented:
Hi @Crazybus, I read the new commit but I don't understand [these lines](https://github.com/elastic/ansible-elasticsearch/blob/486717bc6f4936c244a6c9497f2f8686cab289dd/tasks/elasticsearch-RedHat.yml#L21).

Crazybus commented:
> Hi @Crazybus, I read the new commit but I don't understand these lines.

@fedelemantuano Good because it was wrong :P. I have fixed it up to remove the file instead of adding it (again). I also added some tests to make sure this is actually working properly. After adding the tests I found out that the Debian version was also only removing the package and not the repo so I have fixed that up too!

fedelemantuano commented:
Yes I saw the Debian part, so I'm fixing my [ansible-kibana](https://github.com/fedelemantuano/ansible-kibana) role too.
